src/main/java/games/strategy/io/FileUtils.java

```diff
@@ -0,0 +1,34 @@
+package games.strategy.io;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+/**
+ * A collection of useful methods related to files.
+ */
+public final class FileUtils {
+  private FileUtils() {}
+
+  /**
+   * Returns a collection of abstract pathnames denoting the files and directories in the specified directory.
+   *
+   * @param directory The directory whose files are to be listed.
+   *
+   * @return An immutable collection of files. If {@code directory} does not denote a directory, the collection will be
+   *         empty.
+   *
+   * @see File#listFiles()
+   */
+  public static Collection<File> listFiles(final File directory) {
+    checkNotNull(directory);
+
+    final @Nullable File[] files = directory.listFiles();
+    return files != null ? Collections.unmodifiableList(Arrays.asList(files)) : Collections.emptyList();
```

Comment:
I was mirroring the Apache `FileUtils#listFiles()` signature, which returns `Collection<File>`.  I did considered returning `Iterable<File>`, but this breaks some of the existing code that attempts to stream the result (and I didn't want to add a wrapper around the `Iterable`'s spliterator at the call site just to keep parity).

BTW, while arrays are "iterable" in the sense that they can be used in a for-each loop, I don't think they actually implement `Iterable`.  For example, an array doesn't provide `Iterable` methods such as `forEach()` and `iterator()`.

Reply:
Ok, makes sense.

> BTW, while arrays are "iterable" in the sense that they can be used in a for-each loop, I don't think they actually implement `Iterable`.  For example, an array doesn't provide `Iterable` methods such as `forEach()` and `iterator()`.

Yes, of course, but that's why I put it in quotes, just because I like to think about it this way.

Regarding Iterable, we might want to migrate the TripleA codebase to use `java.nio.file.Path` instead of `java.io.File` simply because it for example implements `Iterable<Path>`, can be converted to a `File` using the `Path#toFile` method and provides a lot of other convinience methods and also support the "user folder symbol" `~`

Reply:
Agreed about preferring `Path` over `File`.  I did consider that here, but decided I didn't want to investigate the nuances of _not_ using `File#listFiles()` in our code at this time. :smile: